### PR TITLE
Modernise and clarify the lhcb development lesson

### PR DESCRIPTION
**Pull request by lbel**
I've merged the lb-git lesson from second-analysis-steps into this one (that was more convenient than the other way around), and shuffled the content around a bit. It now features some general Cern git steps, lb-git, some notes on documentation, and finally a short section on how to use vanilla git with LHCb software. I plan to teach this lesson at the Impactkit.

### CODE REVIEW

File: lhcb-dev.md [hunk 1]
```diff
@@ -137,23 +208,88 @@ create a new merge request, for example https://gitlab.cern.ch/lhcb/Stripping/me
...
+To get an idea of how a certain component of the LHCb software works, you can
+also access the doxygen documentation. One set of doxygen web pages is
+generated for several related projects, and is linked in all the projects web
+sites, like
+[https://cern.ch/LHCb-release-area/DOC/davinci/](https://cern.ch/LHCb-release-area/DOC/davinci/).
+See also the the [LHCb Computing web page](http://cern.ch/lhcb-comp/) for a
+list of projects.
+
+## Working with a full project checkout
+
```
Comment (saschastahl) [inline lhcb-dev.md]: Here we should point to https://gitlab.cern.ch/lhcb/upgrade-hackathon-setup  or
to @rmatev development to make working with project checkouts more convenient
https://gitlab.cern.ch/lhcb-HLT/trigger-dev.

**Comment by saschastahl** [comment]
Can we point out that the LHCb git commands should only be used for very simple changes, you want to push to the software stack, or when you make some development for yourself or your analysis group only? Having seen many merge requests which went wrong using these commands, I think we should encourage the use of project checkouts + plain git.

File: lhcb-dev.md (hunk 1 above)
Comment (lbel) [reply lhcb-dev.md]: Good idea! Can both methods be considered "standard" LHCb tools? I could just add both.

**Comment by lbel** [comment]
I'm definitely going to say that very clearly tomorrow, the text could probably be more explicit about it.

File: lhcb-dev.md (hunk 1 above)
Comment (saschastahl) [reply lhcb-dev.md]: Let's say we would like to make them standard.

**Pull request merged**